### PR TITLE
Dafilesrv interpreter

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: roxie/ccd/ccdquery.cpp [hunk 1]
```diff
@@ -88,7 +93,53 @@ class CQueryDll : implements IQueryDll, public CInterface
     CQueryDll(const char *_dllName, ILoadedDllEntry *_dll) : dllName(_dllName), dll(_dll)
     {
         StringBuffer wuXML;
-        if (!selfTestMode && getEmbeddedWorkUnitXML(dll, wuXML))
+        if (getEmbeddedGraphXML(dll, wuXML))
```
Comment (ghalliday) [inline roxie/ccd/ccdquery.cpp]: The lost test of selTestMode means this will crash

File: roxie/ccd/ccdquery.cpp [hunk 2]
```diff
@@ -88,7 +93,53 @@ class CQueryDll : implements IQueryDll, public CInterface
...
+            {
+                assert(streq(graph->queryName(), "node"));
+                graph->setPropInt("@id", 2);
+                if (!graph->hasProp("hint[@name='dynamic']"))
+                    graph->addPropTree("hint", createPTreeFromXMLString("<hint name='dynamic' value='1'/>"));
+                const char *kindName = graph->queryProp("att[@name='_kind']/@value");
+                ThorActivityKind kind;
+                if (!kindName)
+                    kind = TAKdiskread;
+                else if (strieq(kindName, "DiskRead"))
```
Comment (ghalliday) [inline roxie/ccd/ccdquery.cpp]: Should probably use a function to map names ->kinds

File: rtl/eclrtl/eclhelper_dyn.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        inrec = &in->queryRecordAccessor(true);
+        numOffsets = inrec->getNumVarFields() + 1;
+        translator.setown(createRecordTranslator(queryOutputMeta()->queryRecordAccessor(true), *inrec));
+        if (xgmml.hasProp("att[@name=\"keyfilter\"]"))
+            flags |= TDRkeyed;
+    }
+    virtual bool needTransform() override
+    {
+        return true;
+        //return translator->needsTranslate();
```
Comment (ghalliday) [inline rtl/eclrtl/eclhelper_dyn.cpp]: should this be removed, or does this case need more work.

File: rtl/eclrtl/eclhelper_dyn.cpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    }
+    virtual unsigned getFlags() override
+    {
+        return flags;
+    }
+    virtual void createSegmentMonitors(IIndexReadContext *irc) override
+    {
+        size_t * variableOffsets = (size_t *)alloca(numOffsets * sizeof(size_t));
```
Comment (ghalliday) [inline rtl/eclrtl/eclhelper_dyn.cpp]: Should this be the number of variable size offsets?

File: roxie/ccd/ccdquery.cpp (hunk 2 above)
Comment (richardkchapman) [reply roxie/ccd/ccdquery.cpp]: There's only likely to be 3 or 4 of the 70-odd that we want to support - so more efficient this way.

File: roxie/ccd/ccdquery.cpp (hunk 1 above)
Comment (richardkchapman) [reply roxie/ccd/ccdquery.cpp]: Fixed

File: rtl/eclrtl/eclhelper_dyn.cpp (hunk 3 above)
Comment (richardkchapman) [reply rtl/eclrtl/eclhelper_dyn.cpp]: removed

File: rtl/eclrtl/eclhelper_dyn.cpp (hunk 4 above)
Comment (richardkchapman) [reply rtl/eclrtl/eclhelper_dyn.cpp]: It is:
        numOffsets = inrec->getNumVarFields() + 1;

variable name should probably be improved

File: rtl/eclrtl/eclhelper_dyn.cpp [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            const char *epos = strchr(curFilter,'=');
+            assertex(epos);
+            StringBuffer fieldName(epos-curFilter, curFilter);
+            curFilter = epos+1;
+            assertex (*curFilter == '\'');
+            curFilter++;
+            epos = strchr(curFilter, '\'');
+            StringBuffer fieldVal(epos-curFilter, curFilter);
+            unsigned fieldNum = inrec->getFieldNum(fieldName);
+            assertex(fieldNum != -1);
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply rtl/eclrtl/eclhelper_dyn.cpp]: fixed

File: rtl/eclrtl/rtlrecord.cpp [hunk 6]
```diff
@@ -443,6 +443,15 @@ const RtlRecord &COutputMetaData::queryRecordAccessor(bool expand) const
     return *useAccessor;
 }
 
+size32_t COutputMetaData::getRecordSize(const void * data)
+{
+    const RtlRecord &r = queryRecordAccessor(false);
+    unsigned numOffsets = r.getNumVarFields() + 1;
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply rtl/eclrtl/rtlrecord.cpp]: or after...

**Pull request closed without merging**